analytics/client.py

```diff
@@ -34,6 +35,12 @@ def __init__(self, write_key=None, debug=False, max_queue_size=10000,
         self.debug = debug
         self.send = send
 
+        # On program exit, allow the consumer thread to exit cleanly.
+        # This prevents exceptions and a messy shutdown when the interpreter is
+        # destroyed before the daemon thread finishes execution. However, it
+        # is *not* the same as flushing the queue! To guarantee all
+        atexit.register(self.join)
```

Comment:
How about just not making it a daemon thread? 
If that doesn't work can you leave a comment explaining why so the next maintainer can get more understanding?

Also it looks like the comment here is incomplete.